packages/shared/dao/index.ts

```diff
@@ -42,64 +42,92 @@ export async function fetchCatalystRealms(nodesEndpoint: string | undefined): Pr
   return nodes
 }
 
-export function isPeerHealthy(peerStatus: Record<string, HealthStatus>) {
-  return (
-    Object.keys(peerStatus).length > 0 &&
-    !Object.keys(peerStatus).some((server) => {
-      return peerStatus[server] !== HealthStatus.HEALTHY
-    })
-  )
-}
-
-export function lambdasHealthUrl(domain: string) {
-  return `${domain}/lambdas/health`
-}
+export async function fetchCatalystStatus(
+  domain: string,
+  denylistedCatalysts: string[]
+): Promise<Candidate | undefined> {
+  if (denylistedCatalysts.includes(domain)) return undefined
+
+  const [aboutResponse, parcelsResponse] = await Promise.all([ask(`${domain}/about`), ask(`${domain}/stats/parcels`)])
+
+  if (aboutResponse.httpStatus !== 404) {
+    const result = aboutResponse.result
+    if (
+      aboutResponse.status === ServerConnectionStatus.OK &&
+      result &&
+      result.comms &&
+      result.configurations &&
+      result.bff
+    ) {
+      const { comms, configurations, bff } = result
+
+      // TODO(hugo): this is kind of hacky, the original representation is much better,
+      // but I don't want to change the whole pick-realm algorithm now
+      const usersParcels: Parcel[] = []
+
+      if (parcelsResponse.result && parcelsResponse.result.parcels) {
+        for (const {
+          peersCount,
+          parcel: { x, y }
+        } of parcelsResponse.result.parcels) {
+          const parcel: Parcel = [x, y]
+          for (let i = 0; i < peersCount; i++) {
+            usersParcels.push(parcel)
+          }
+        }
+      }
 
-export function commsStatusUrl(domain: string, includeUsersParcels: boolean = false) {
-  let url = `${domain}/comms/status`
-  const queryParameters: string[] = []
+      return {
+        protocol: comms.protocol,
+        catalystName: configurations.realmName,
+        domain: domain,
+        status: aboutResponse.status,
+        elapsed: aboutResponse.elapsed!,
+        usersCount: bff.userCount ?? 0,
+        maxUsers: -1, // TODO
```

Comment:
Can we refer this to an issue or have more context in the comment?

Reply:
Sorry, I'll remove the comment, we are not going to use maxUsers anymore, since now we have the feature flag for disabled catalyst